src/test/java/SalesforceModules/OppurtunitiesUtil.java

```diff
@@ -188,93 +193,109 @@ public static void backToAccount(String logStep) throws InterruptedException {
     /**
      * Open the Opportunity
      *
-     *
      * @param logStep the log
      */
     public static void openOpportunity(String logStep) throws InterruptedException {
         KeywordUtil.delay(2000);
         KeywordUtil.waitForVisible(OppurtunitiesObject.selectOpportunity);
-        KeywordUtil.click(OppurtunitiesObject.selectOpportunity,logStep);
+        KeywordUtil.click(OppurtunitiesObject.selectOpportunity, logStep);
     }
 
     /**
      * Select the Stage - Proposal in Quote
      *
-     *
      * @param logStep the log
      */
-    public static void selectStageProposalQuote(String logStep){
+    public static void selectStageProposalQuote(String logStep) {
         try {
+            KeywordUtil.delay(2000);
             KeywordUtil.waitForVisible(OppurtunitiesObject.proposalStage);
             KeywordUtil.click(OppurtunitiesObject.proposalStage, logStep);
-        }catch (Exception e) {
+        } catch (Exception e) {
             KeywordUtil.waitForVisible(OppurtunitiesObject.proposalStage);
-            WebElement element = KeywordUtil.getDriver().findElement(OppurtunitiesObject.proposalStage);
-            JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
-            executor.executeScript("arguments[0].click();", element);
+            KeywordUtil.clickJS(OppurtunitiesObject.proposalStage);
+        }
+    }
+    public static void selectStageProposalQuotes(String logStep) {
+        try {
+            KeywordUtil.delay(2000);
+            KeywordUtil.waitForVisible(OppurtunitiesObject.proposalStages);
+            KeywordUtil.click(OppurtunitiesObject.proposalStages, logStep);
+        } catch (Exception e) {
+            KeywordUtil.waitForVisible(OppurtunitiesObject.proposalStages);
+            KeywordUtil.clickJS(OppurtunitiesObject.proposalStages);
         }
     }
-
     /**
      * Clicking mark as stage button
      *
-     *
      * @param logStep the log
      */
-    public static void setMarkAsStage(String logStep){
+    public static void setMarkAsStage(String logStep) throws InterruptedException {
         try {
+            KeywordUtil.delay(3000);
             KeywordUtil.waitForVisible(OppurtunitiesObject.markStage);
             KeywordUtil.click(OppurtunitiesObject.markStage, logStep);
-        }catch (Exception e) {
+        } catch (Exception e) {
+            KeywordUtil.delay(3000);
             KeywordUtil.waitForVisible(OppurtunitiesObject.markStage);
             WebElement element = KeywordUtil.getDriver().findElement(OppurtunitiesObject.markStage);
             JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
             executor.executeScript("arguments[0].click();", element);
         }
     }
+    public static void setMarkAsStages(String logStep) throws InterruptedException {
```

Comment:
Please add the comment for the function description

Reply:
Added comments